src/system_identity/generator.rs

```diff
@@ -0,0 +1,62 @@
+use thiserror::Error;
+
+use crate::{key::creator::Creator as KeyCreator, TokenRetriever};
+
+use super::iam_client::{IAMClient, SystemIdentityCreationResponseData};
+
+#[derive(Debug, Clone, Error)]
+pub enum SystemIdentityCreationError {
+    // This should probably leverage associated types for the involved traits, so we
+    // can gain automatic conversions via #[from] annotations...
+    #[error("error retrieving token: `{0}`")]
+    TokenRetriever(String),
+    #[error("error creating key pair: `{0}`")]
+    KeyPairCreator(String),
+    #[error("error creating system identity: `{0}`")]
+    IAMClient(String),
+}
```

Comment:
Nit: should we use the term 'generation' instead of 'creation' (that would match the module name)

Reply:
Done in https://github.com/newrelic/newrelic-auth-rs/pull/25/commits/99c6295491423597116e78fda0994897dbc271f4!